backon/src/backoff/mod.rs

```diff
@@ -12,3 +12,18 @@ pub use fibonacci::FibonacciBuilder;
 mod exponential;
 pub use exponential::ExponentialBackoff;
 pub use exponential::ExponentialBuilder;
+
+#[cfg(feature = "std")]
+fn f32() -> f32 {
+    fastrand::f32()
+}
+
+#[cfg(all(not(feature = "std"), feature = "embassy-time"))]
+fn f32() -> f32 {
+    fastrand::Rng::with_seed(embassy_time::Instant::now().as_micros()).f32()
+}
+
+#[cfg(all(not(feature = "std"), not(feature = "embassy-time")))]
+fn f32() -> f32 {
+    fastrand::Rng::with_seed(0x2fdb0020ffc7722b).f32()
```

Comment:
How about adding a seed option on `ExponentialBuilder` instead? In this way, we can avoid unexpected behavior happened.

We can't carry an `Rng` in builder since it doesn't copy.

Reply:
Sounds good to me.